cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/BasicStreamWriterSizeCollector.java

```diff
@@ -43,6 +67,16 @@ public synchronized void received(String streamName, long dataSize) {
     AtomicLong value = streamSizes.get(streamName);
     if (value == null) {
       value = streamSizes.putIfAbsent(streamName, new AtomicLong(dataSize));
+      if (value != null) {
+        // This is the first time that we've seen this stream, we subscribe to generation changes
+        truncationSubscriptions.add(streamCoordinatorClient.addListener(streamName, new StreamPropertyListener() {
```

Comment:
If we replace the aggregator with a new one in the ``aggregators`` map, inside the truncation listener, like you set above, then we don't need that, right?